shared/src/http.rs

```diff
@@ -131,6 +135,98 @@ pub fn filter_hop_by_hop(headers: &mut HeaderMap, version: Version) -> &mut Head
     headers
 }
 
+/// Send a request to a single upstream with configurable timeout
+pub async fn send_to_upstream<C, B, E>(
+    client: &Client<C, B>,
+    upstream_url: &url::Url,
+    request: Request<B>,
+    timeout_secs: u64,
+) -> Result<Response<Bytes>, E>
+where
+    C: hyper_util::client::legacy::connect::Connect + Clone + Send + Sync + 'static,
+    B: hyper::body::Body + Send + Unpin + 'static,
+    B::Data: Send,
+    B::Error: std::error::Error + Send + Sync + 'static,
+    E: From<UpstreamError>,
+{
+    // Use host as identifier for error messages
+    let upstream_identifier = upstream_url.host_str().unwrap_or(upstream_url.as_str());
+
+    // Build the full upstream URI by combining base URL with request path
+    let path_and_query = request
+        .uri()
+        .path_and_query()
+        .map(|pq| pq.as_str())
+        .unwrap_or("/");
+
+    let mut url = upstream_url.clone();
+    if let Some((path, query)) = path_and_query.split_once('?') {
+        url.set_path(path);
+        url.set_query(Some(query));
+    } else {
+        url.set_path(path_and_query);
+    }
+    let upstream_uri = url.to_string();
+
+    // Build request to send to upstream with modified URI and filtered headers
+    let (mut parts, body) = request.into_parts();
+    filter_hop_by_hop(&mut parts.headers, parts.version);
```

Comment:
I see that the send to upstream version used by the ingest router does hop by hop filtering in one direction (client -> upstream), but doesn't add the via header. The proxy does hop by hop filtering on both request and request and also adds the via header in both cases.

Is that difference intentional? If yes, how come this function is in the shared module?